tables/TaQL/TableParse.cc

```diff
@@ -242,155 +252,162 @@ TableParseSelect::~TableParseSelect()
+    }
+  }
+  // Split the first subtable name into parts using a dot as separator.
+  // The first part can be empty, a shorthand or a temporary table number.
+  // An empty part means the first available table.
+  stPart = 1;          // indicate first part is handled.
+  Vector<String> parts = stringToVector(subs[0], '.');
+  if (parts.size() == 0  ||  parts[0].empty()) {
+    table = findTable (String(), True, stack);
+    if (table.isNull()) {
+      throw TableInvExpr(":: or . is invalid in table name " + name +
+                         ": no previous table available");
     }
-  } else if (! ftab.isNull()) {
-    //# The table is a temporary table (from a select clause).
-    table = ftab;
   } else {
-    //# The table name is a string.
-    //# If the name contains ::, it is a table keyword in a table at an outer
-    //# SELECT statement.
-    String shand, columnName;
-    Vector<String> fieldNames;
-    if (splitName (shand, columnName, fieldNames, name, False, False, True)) {
-      table = tableKey (name, shand, columnName, fieldNames, stack);
+    if (tabnr >= 0) {
+      // Temporary table number (1-based) given.
+      if (tabnr < 1  ||  tabnr > Int(tempTables.size())
+          ||  tempTables[tabnr-1] == 0) {
+        throw (TableInvExpr ("Invalid temporary table number given in " + name));
+      }
+      table = *(tempTables[tabnr-1]);
     } else {
-      // If no or equal shorthand is given, try to see if the
-      // given name is already used as a shorthand (also in the WITH tables).
-      // If so, take the table of that shorthand.
-      Bool foundSH = False;
-      if (shorthand.empty()  ||  name == shorthand) {
-        for (Int i=stack.size()-1; i>=0; i--) {
-          Table tab = stack[i]->findTable (name, True);
-          if (! tab.isNull()) {
-            table = tab;
-            foundSH = True;
-            break;
-          }
+      // See if the first part is a shorthand.
+      table = findTable (parts[0], True, stack);
+      if (table.isNull()) {
+        // It was not something like shorthand.column, thus try as a full name.
+        // However, do not open if alwaysOpen=False.
+        // In that case the table is opened when needed
+        // (because the name can contain a wildcard as well).
+        if (!alwaysOpen) {
+          return table;
+        }
+        stPart = 0;
+        stSub = 1;
+        table = Table (subs[0]);
+        if (table.isNull()) {
+          throw TableInvExpr("Table " + subs[0] + " is unknown");
         }
       }
-      if (!foundSH  &&  alwaysOpen) {
-        table = TableUtil::openTable(name);
+    }
+  }
+  // Okay; we have found the first table.
+  AlwaysAssert (!table.isNull(), AipsError);
+  // Now process all parts in all subtable names, where the first name or
+  // first part might need to be skipped because already processed.
+  const TableRecord* keywords = &(table.keywordSet());
+  for (uInt k=stSub; k<subs.size(); ++k) {
```

Comment:
Is there a reason to use `Int` and `uInt` in new code?

Reply:
Not really. Those are the casacore types which I'm pretty used to.